pkg/kubernetes/dynamicInformers.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"context"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/informers"
+	restclient "k8s.io/client-go/rest"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+var (
+	dynamicInformerFactoryMap           = make(map[string]dynamicinformer.DynamicSharedInformerFactory)
+	dynamicInformerInitLock             = &sync.Mutex{}
+	supervisorDynamicInformerFactoryMap = make(map[string]dynamicinformer.DynamicSharedInformerFactory)
+	supervisorDynamicInformerInitLock   = &sync.Mutex{}
+)
+
+// newDynamicInformerFactory creates a dynamic informer factory for a given namespace if it doesn't exist already
+func newDynamicInformerFactory(ctx context.Context, cfg *restclient.Config, namespace string, isInCluster bool) (dynamicinformer.DynamicSharedInformerFactory, error) {
+	log := logger.GetLogger(ctx)
+
+	// Acquire lock
+	if isInCluster {
```

Comment:
How about clubbing this if-else with the one below?

Reply:
Done